intellij-ext/modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt

```diff
@@ -56,20 +56,18 @@ class MirrordConfigDropDown : ComboBoxAction() {
         e.project?.let { project ->
             // this check ensures that we don't query the index when it is being built
             // querying the index during the startup can give us 0
-            if (DumbService.isDumb(project)) {
-                e.presentation.isVisible = false
-                return
-            }
-
             if (!::configFiles.isInitialized) {
-                val basePath = project.basePath ?: throw Error("couldn't resolve project path")
-                configFiles = FileBasedIndex.getInstance().getAllKeys(MirrordConfigIndex.key, project).filter {
-                    it.startsWith(basePath)
-                }.toHashSet()
+                // there is a possibility of a race condition in case index is accessed while being built
+                // so all index queries need to be done in smart mode
+                runInSmartModeOrDirectly({
+                    initializeConfigs(project)
+                }, project)
```

Comment:
Nitpick: I think it would be prettier to have `runInSmartModeOrDirectly` take a function that takes a project parameter, so that you can call it like this: `runInSmartModeOrDirectly(initializeConfigs, project)`
(and then in runInSmartModeOrDirectly call `action(project)` instead of `action()`).